include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -674,26 +676,27 @@ class Radiation : public ISimulationPlugin
       hdf5DataFile.open(filename.str().c_str(), fAttr);
 
       typename PICToSplash<float_64>::type radSplashType;
+      typename PICToSplash<float_32>::type radDirSplashType;
+      typename PICToSplash<float_32>::type radFreqSplashType;
```

Comment:
@psychocoderHPC  This is the statement I mean. I can replace `float_64` here by `Amplitude::complex_T::type`, thereby making the precision dependent on the default precision of the Amplitude's components.

Doing the same for the direction vectors and frequencies could be accomplished, as you suggest, by a typedef for the entire module (i.e., in `VectorTypes.hpp`). I was thinking more along the lines of inferring the type at compile time from the underlying float type of the pointers in ll.132,133 (`detectorPositions, detectorFrequencies`).

Reply:
`PICToSplash` supports all types defined [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/454a28efe6a6eb0fa1e0e97031ef2569a1ad313b/include/picongpu/traits/PICToSplash.tpp#L36)

>  I was thinking more along the lines of inferring the type at compile time from the underlying float type of the pointers in ll.132,133 (detectorPositions, detectorFrequencies).

Also possible but I do not know if this would be more complex. For this decision, I have not enough overview about this code.